packages/rest/src/controllers/proofs/ProofController.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  Agent,
+  JsonTransformer,
+  PresentationPreview,
+  RecordNotFoundError,
+  RequestedCredentials,
+  IndyCredentialInfo,
+} from "@aries-framework/core";
+import { JsonEncoder } from "@aries-framework/core/build/utils/JsonEncoder";
+import {
+  JsonController,
+  Param,
+  Get,
+  NotFoundError,
+  Body,
+  Post,
+  InternalServerError,
+  Delete,
+} from "routing-controllers";
+import { Service, Inject } from "typedi";
+
+import { AcceptProofProposalRequest } from "../../schemas/AcceptProofProposalRequest";
+import { ProofPresentationRequest } from "../../schemas/ProofPresentationRequest";
+import { ProofProposelRequest } from "../../schemas/ProofProposelRequest";
+import { SendProofPresentationRequest } from "../../schemas/SendProofPresentationRequest";
+import { CredentialUtils } from "../../utils/CredentialUtils";
+import { ProofUtils } from "../../utils/ProofUtils";
+
+@JsonController("/proofs")
+@Service()
+export class ProofController {
+  @Inject()
+  private agent: Agent;
+  private proofUtils: ProofUtils;
+
+  public constructor(agent: Agent) {
+    this.agent = agent;
+    this.proofUtils = new ProofUtils(agent);
+  }
+
+  /**
+   * Retrieve proof record by proofRecordId
+   */
+  @Get("/:proofRecordId")
+  public async getProofById(@Param("proofRecordId") proofRecordId: string) {
+    const proofRecord = await this.agent.proofs.getById(proofRecordId);
+
+    if (!proofRecord) {
+      throw new NotFoundError(`proof record with proofRecordId "${proofRecordId}" not found.`);
+    }
+
+    return proofRecord;
+  }
+
+  /**
+   * Retrieve proof record by threadId
+   */
+  @Get("/thread/:threadId")
+  public async getProofByThreadId(@Param("threadId") threadId: string) {
+    const proofRecord = await this.proofUtils.getProofByThreadId(threadId);
```

Comment:
This will throw an error if not found I think.

The convention in AFJ is
- getXXX -> throws error if not found
- findXXX -> returns null if not found